lib/HLSL/DxilValidation.cpp

```diff
@@ -3523,6 +3526,23 @@ static void ValidateSignature(ValidationContext &ValCtx, const DxilSignature &S,
       }
       DepthKind = semanticKind;
       break;
+    case DXIL::SemanticKind::Barycentrics: {
+      // There can only be up to two SV_Barycentrics
+      // with differeent perspective interpolation modes.
+      if (baryInterpolationModes.size() > 1) {
+        ValCtx.EmitError(ValidationRule::MetaBarycentricsTwoPerspectives);
```

Comment:
Falls through to possible duplicate error - one should be enough.  Either break here, or provide distinct errors for more than two and multiple with same perspective.